src/memory.rs

```diff
@@ -54,100 +54,138 @@ impl ByteSliceView {
     }
 }
 
-#[no_mangle]
-pub extern "C" fn allocate_rust(ptr: *const u8, length: usize) -> Buffer {
-    // Go doesn't store empty buffers the same way Rust stores empty slices (with NonNull  pointers
-    // equal to the offset of the type, which would be equal to 1 in this case)
-    // so when it wants to represent an empty buffer, it passes a null pointer with 0 length here.
-    if length == 0 {
-        Buffer::from_vec(Vec::new())
-    } else {
-        Buffer::from_vec(Vec::from(unsafe { slice::from_raw_parts(ptr, length) }))
-    }
+/// A view into a `Option<&[u8]>`, created and maintained by Rust.
+///
+/// This can be copied into a []byte in Go.
+#[repr(C)]
```

Comment:
This is the counterpart to ByteSliceView for Rust->Go. Makes sense

Reply:
Right. In Rust we could use a non-optional byte slice `&[u8]`. But since this becomes optional once converted to the Go type, I thought it is better to add first-class optional support on the Rust side as well.

If you ever forget which one is which: A view with `is_nil` looks into Go memory. A view with `is_none` looks into Rust memory.

Reply:
Good naming choices to match the languages